tez-mapreduce/src/main/java/org/apache/hadoop/mapred/split/TezGroupedSplitsInputFormat.java

```diff
@@ -193,8 +236,8 @@ protected boolean initNextRecordReader() throws IOException {
 
       // get a record reader for the idx-th chunk
       try {
-        curReader = wrappedInputFormat.getRecordReader(
-            groupedSplit.wrappedSplits.get(idx), job, reporter);
+        curReader = initedReaders.poll().get();
+        submitInitReaders(1);
```

Comment:
From the next stage onwards (i.e after the init part), this will start executing in sequential order as it requests for 1 additional reader. It will be good to init next set of readers in parallel.